core/bridges/bridge_cache.go

```diff
@@ -0,0 +1,261 @@
+package bridges
+
+import (
+	"context"
+	"database/sql"
+	"errors"
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/smartcontractkit/chainlink-common/pkg/services"
+	"github.com/smartcontractkit/chainlink-common/pkg/sqlutil"
+
+	"github.com/smartcontractkit/chainlink/v2/core/logger"
+	"github.com/smartcontractkit/chainlink/v2/core/utils"
+)
+
+const (
+	BridgeCacheServiceName   = "BridgeCache"
+	DefaultUpsertInterval    = 500 * time.Millisecond
+	DefaultBulkInsertTimeout = 250 * time.Millisecond
+)
+
+type BridgeCache struct {
+	// dependencies and configurations
+	ORM
+	lggr     logger.Logger
+	interval time.Duration
+
+	// service state
+	services.StateMachine
+	wg   sync.WaitGroup
+	stop chan struct{}
+
+	// data state
+	bridgeTypesCache     sync.Map
+	bridgeLastValueCache map[string]BridgeResponse
+	mu                   sync.RWMutex
+}
+
+var _ ORM = (*BridgeCache)(nil)
+var _ services.Service = (*BridgeCache)(nil)
+
+func NewBridgeCache(base ORM, lggr logger.Logger, upsertInterval time.Duration) *BridgeCache {
```

Comment:
Could shorten to just Cache since we are in package bridges already:
```suggestion
func NewCache(base ORM, lggr logger.Logger, upsertInterval time.Duration) *Cache {
```